### PR TITLE
Migrate fuchsia_precache to shard tests.

### PR DESCRIPTION
**Pull request by godofredoc**
Adhoc tests are being deprecated and existing tests are being migrated to shard tests.

Bug: https://github.com/flutter/flutter/issues/139153

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by godofredoc** [comment]
Led run: https://chromium-swarm.appspot.com/task?id=66349138641eef10

File: dev/bots/test.dart [hunk 1]
```diff
@@ -1602,6 +1603,41 @@ Future<void> _runAnalyze() async {
...
+      '--no-ios',
+      '--force',
+    ],
+    workingDirectory: flutterRoot,
+  );
+  await runCommand(
+    'flutter',
+    <String>[
+      'precache',
+      '--flutter_runner',
```
Comment (goderbauer) [inline dev/bots/test.dart:1632]: could this just be combined with the previous `flutter precache` command?

**Review by goderbauer — APPROVED** [review]
LGTM

File: dev/bots/test.dart (hunk 1 above)
Comment (christopherfujino) [reply dev/bots/test.dart:1632]: Yes, I don't see why not (unless there's some hidden gotcha in the flutter precache command I'm not aware of, in which case we should just fix it)

Comment (godofredoc) [reply dev/bots/test.dart:1632]: ok, sending an update.

Comment (godofredoc) [reply dev/bots/test.dart:1621]: Led run with the new changes: https://chromium-swarm.appspot.com/task?id=6635f944ed3c5810

**Pull request merged**